### DISCUSSION TITLE
Don't pollute /usr/local

### DISCUSSION THREAD

**Discussion by ethomson** [community/community#78266] [Actions]
### Select Topic Area

Product Feedback

### Body

Your latest image update broke my workflow runs.

Your newest image update causes my builds to start failing because it introduced `/usr/local/bin/python`. 

Please don't pre-install things into `/usr/local/`. This is where (eg) brew installs things.

My build now fails with:

```
==> Pouring python@3.11--3.11.6_1.monterey.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/2to3
Target /usr/local/bin/2to3
already exists. You may want to remove it:
  rm '/usr/local/bin/2to3'

To force the link and overwrite all conflicting files:
  brew link --overwrite python@3.11

To list all files that would be deleted:
  brew link --overwrite --dry-run python@3.11

Possible conflicting files are:
/usr/local/bin/2to3 -> /Library/Frameworks/Python.framework/Versions/3.12/bin/2to3
/usr/local/bin/2to3-3.11 -> /Library/Frameworks/Python.framework/Versions/3.11/bin/2to3-3.11
/usr/local/bin/idle3 -> /Library/Frameworks/Python.framework/Versions/3.12/bin/idle3
/usr/local/bin/idle3.11 -> /Library/Frameworks/Python.framework/Versions/3.11/bin/idle3.11
/usr/local/bin/pydoc3 -> /Library/Frameworks/Python.framework/Versions/3.12/bin/pydoc3
/usr/local/bin/pydoc3.11 -> /Library/Frameworks/Python.framework/Versions/3.11/bin/pydoc3.11
/usr/local/bin/python3 -> /Library/Frameworks/Python.framework/Versions/3.12/bin/python3
/usr/local/bin/python3-config -> /Library/Frameworks/Python.framework/Versions/3.12/bin/python3-config
/usr/local/bin/python3.11 -> /Library/Frameworks/Python.framework/Versions/3.11/bin/python3.11
/usr/local/bin/python3.11-config -> /Library/Frameworks/Python.framework/Versions/3.11/bin/python3.11-config
```

You shouldn't be installing things into `/usr/local`. That's for _me_.

To quote from the Linux FHS:

> The `/usr/local` hierarchy is for use by the system administrator when installing software locally. It needs to be safe from being overwritten when the system software is updated. It may be used for programs and data that are shareable amongst a group of hosts, but not found in `/usr`.

In case there's confusion: in this case _I_ am the system administrator, you are providing a distribution for me. `/usr/local` should be _empty_ when my workflow is run.

(It's not clear to me why you need to put symlinks to Python in `/usr/local` anyway, macOS itself puts these in `/usr`. This breakage just feels gratuitous. 🥴)

**Comment by Cyd0n1a** [comment]
Agreed.

**Comment by ethomson** [comment]
It's actually a bit more complicated / worse. The image generation (whether explicitly or as a transitive dependency) installed python using brew. Then added symlinks back to the system python. 🥴 

So homebrew is terribly confused about the state of python and can't update it, which means that you can't brew install anything that depends on python. Ouch.

Y'all need a philosophy about `/usr/local` - what do you put in there at image generation time. Ideally nothing! That's for me! But "we install things with homebrew and you can use homebrew to add or remove those things" is a defensible position. Using homebrew AND other package managers or adding random things in there that collide with homebrew is not.
